apps/root/src/services/conctactListService.ts

```diff
@@ -14,83 +21,104 @@ export default class ContactListService implements IContactListService {
     const user = this.accountService.getUser();
     if (!user) {
       return;
     }
-    const currentContacts = [...this.contactList];
+
     try {
       const signature = await this.accountService.getWalletVerifyingSignature({});
-      this.contactList = [...currentContacts, contact];
-      await this.meanApiService.postContacts({ contacts: [contact], accountId: user.id, signature });
+      return await this.meanApiService.getAccountLabelsAndContactList({
+        accountId: user.id,
+        signature,
+      });
     } catch (e) {
-      this.contactList = currentContacts;
-      throw e;
+      console.error(e);
     }
   }
 
-  async removeContact(contact: Contact): Promise<void> {
+  async addContact(contact: Contact): Promise<void> {
     const user = this.accountService.getUser();
     if (!user) {
       return;
     }
     const currentContacts = [...this.contactList];
+    const currentLabel = this.labelService.labels?.[contact.address];
     try {
-      remove(this.contactList, { address: contact.address });
       const signature = await this.accountService.getWalletVerifyingSignature({});
-      await this.meanApiService.deleteContact({ contactAddress: contact.address, accountId: user.id, signature });
+      this.contactList = [...currentContacts, contact];
+      this.labelService.labels = {
+        ...this.labelService.labels,
+        [contact.address]: contact.label?.label
+          ? { label: contact.label.label, lastModified: Date.now() }
+          : currentLabel,
+      };
+      await this.meanApiService.postContacts({ contacts: [contact], accountId: user.id, signature });
     } catch (e) {
       this.contactList = currentContacts;
+      this.labelService.labels = {
+        ...this.labelService.labels,
+        [contact.address]: currentLabel,
+      };
       throw e;
     }
   }
 
-  async editContact(contact: Contact): Promise<void> {
+  async removeContact(contact: Contact): Promise<void> {
     const user = this.accountService.getUser();
-    if (!user || !contact.label) {
-      return;
-    }
-    const contactIndex = findIndex(this.contactList, { address: contact.address });
-    if (contactIndex === -1) {
+    if (!user) {
       return;
     }
-
     const currentContacts = [...this.contactList];
     try {
-      this.contactList[contactIndex] = contact;
+      this.contactList = currentContacts.filter((contactEl) => contactEl.address !== contact.address);
       const signature = await this.accountService.getWalletVerifyingSignature({});
-      await this.meanApiService.putAccountLabel({
-        newLabel: contact.label.label,
-        labeledAddress: contact.address,
-        accountId: user.id,
-        signature,
-      });
+      await this.meanApiService.deleteContact({ contactAddress: contact.address, accountId: user.id, signature });
     } catch (e) {
       this.contactList = currentContacts;
       throw e;
     }
   }
 
-  getContacts(): ContactList {
```

Comment:
I think we should still keep this, so we don't `bypass` the underlying data logic through the UI